### PR TITLE
Adds no-else-return rule

### LINKED ISSUE #29: no-else-return
https://eslint.org/docs/rules/no-else-return

### PR DESCRIPTION
**Pull request by BeccaAnderton**
## Suggested rule/changes(s):
* `<no-else-return>` 

## Reason for addition/amendment
* Closes #29 
* Disallows return statements prior to an else if or else statement
* See examples of code in readme.md
* Also doesn't allow else statements - sorry boys!

@netsells/frontend - Please review

### CODE REVIEW

**Comment by rowbi-wan** [comment]
Me no liek. why cant I return?

**Comment by BeccaAnderton** [comment]
@rowancollins89 you can. Just not before an else / else if. Else / else if's a gross anyway and we should be avoiding them wherever possible. There's also no point returning in an else because it would be the last thing to return anyway.

**Comment by rowbi-wan** [comment]
its past my bed time. I'm being thick.
```
if(foo !== bar) {
    return foo;
}
return bar;
```

would work so i'm happy.

**Comment by BeccaAnderton** [comment]
@rowancollins89 you are correct - although you'd need a space after that if 😉

**Review by samturrell — APPROVED** [review]

File: rules/javascript.js [hunk 1]
```diff
@@ -25,5 +25,8 @@ module.exports = {
         }],
         // Discourage using 'var' for creating variables - require using let/const instead
         'no-var': _THROW.ERROR,
+        'no-else-return': [_THROW.WARNING, {
```
Comment (samturrell) [inline rules/javascript.js]: Comment

File: rules/javascript.js [hunk 2]
```diff
@@ -25,5 +25,8 @@ module.exports = {
...
+            allowElseIf: false,
+        }]
```
Comment (samturrell) [inline rules/javascript.js]: Trailing comma pls

**Review by samturrell — CHANGES_REQUESTED** [review]

**Comment by BeccaAnderton** [comment]
@samturrell fixed these!

**Comment by samturrell** [comment]
doesn't look liek it

**Comment by BeccaAnderton** [comment]
@samturrell I did. But you'd pushed to this branch last night. So my push failed. Check again.

**Review by samturrell — APPROVED** [review]

**Pull request merged**